src/index.js

```diff
@@ -108,6 +115,38 @@ class Saturn {
     return { res, log }
   }
 
+  async * fetchContentWithFallback (cidPath, opts = {}, origins) {
+    if (this.nodes.length === 0) {
+      await this.loadNodesPromise
+    }
+    let lastError = null
+    // we use this to checkpoint at which chunk a request failed.
+    // this is temporary until range requests are supported.
+    let byteCountCheckpoint = 0
+    for (const origin of this.nodes) {
```

Comment:
Retrying 100 times seems a bit overkill, especially if the error is consistent and will throw on every attempt. Like there's something wrong with the DAG itself and not the L1. I think we should start smaller at like 10 and see how it goes.

Reply:
yeah agreed, was thinking about that as well. 

Wondering if there are errors that we should handle, like if the js client verification fails due to some js error then it will probably fail for every fallback node regardless

Reply:
I've added a limit to the amount we can fallback. I do think some errors should be bailed right away but i think id save that for another PR. 

Im also worried about perf with error matching mid request. Thoughts?

Reply:
Perf shouldn't be an issue unless the error list is huge, like 10k+.